### PR TITLE
Fix: v4.4.0 fix bugs

### PR DESCRIPTION
**Pull request by MaShizhen**

### CODE REVIEW

File: src/app/flux/printing/index.js [hunk 1]
```diff
@@ -2582,8 +2600,8 @@ export const actions = {
 
         const modelState = modelGroup.selectMultiModel(intersect, selectEvent);
         dispatch(actions.updateState(modelState));
-
-        dispatch(actions.render());
+        // TODO: Performance optimization test
```
Comment (jane-rose) [inline src/app/flux/printing/index.js:2603]: check `todo`

File: src/app/flux/printing/index.js [hunk 2]
```diff
@@ -5002,7 +5058,9 @@ export const actions = {
 export default function reducer(state = INITIAL_STATE, action) {
     switch (action.type) {
         case ACTION_UPDATE_STATE: {
-            return Object.assign({}, state, action.state);
+            const s = Object.assign({}, state, action.state);
+            window.pp = s
```
Comment (jane-rose) [inline src/app/flux/printing/index.js:5062]: remove debug code

File: src/app/lib/buffer-utils.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
+type TPoint = { x: number, y: number }
+
+export const bufferToPoint = (buffer: ArrayBuffer) => {
+    if (buffer instanceof ArrayBuffer) {
+        const view = new DataView(buffer);
+        const points: TPoint[] = [];
```
Comment (jane-rose) [inline src/app/lib/buffer-utils.ts:6]: count the array length at first?

File: src/app/ui/components/Input/TextArea.jsx [hunk 4]
```diff
@@ -14,6 +14,7 @@ const TextAreaInput = React.memo(({
     function onInsideChange(event) {
         if (displayValue !== event.target.value) {
             setDisplayValue(event.target.value);
+            console.log('d', setDisplayValue);
```
Comment (jane-rose) [inline src/app/ui/components/Input/TextArea.jsx:17]: log

File: src/app/ui/pages/MachineMaterialSettings/machineSettings.jsx [hunk 5]
```diff
@@ -160,77 +160,74 @@ const toolHeadMap = {
...
     isDefault: true
 }];
 const defaultNozzleDiameterListForDualExtruderArtisan = [{
-    value: 0.2,
+    value: '0.2',
     label: '0.2',
     isDefault: true
 }, {
-    value: 0.4,
+    value: '0.4',
```
Comment (jane-rose) [inline src/app/ui/pages/MachineMaterialSettings/machineSettings.jsx:189]: recheck this

File: src/app/ui/pages/MachineMaterialSettings/machineSettings.jsx [hunk 6]
```diff
@@ -346,55 +354,43 @@ const MachineSettings = ({
...
 
     const getNozzleDiameterFromStorage = (direction, _currentSerial) => {
-        const key = `customNozzleDiameter.${currentToolHead}.${_currentSerial === MACHINE_SERIES.A400.value ? 'artisan' : '_'}.${direction}`;
+        const key = `customNozzleDiameter.${currentToolHead}.${_currentSerial}.${direction}`;
         const str = machineStore.get(key) || '[]';
         const configs = JSON.parse(str);
         return configs.map((value) => {
             return {
-                isDefault: false,
+                value: `${value}`,
```
Comment (jane-rose) [inline src/app/ui/pages/MachineMaterialSettings/machineSettings.jsx:362]: change string to number

Comment (tamer1an) [reply src/app/ui/pages/MachineMaterialSettings/machineSettings.jsx:362]: otherwise use `String(value)`

**Pull request merged**